spellings/action.yml

```diff
@@ -19,105 +19,105 @@ inputs:
+      sudo apt-get install npm -y
+      sudo npm install -g cspell
+
+      # Add the Github Action Path to PATH
+      export PATH="$GITHUB_ACTION_PATH:$PATH"
+
+      # Account for starting with an alternate path in a repository
+      # Do this by copying the cspell config and wordlist to the desired path
+      # Wrap in a set +e so Github doesn't throw an error if the file or
+      # directory already exists.
+      set +e
+      cp cspell.config.yaml ${{ inputs.path }}
+      mkdir ${{ inputs.path }}/.github
+      cp .github/.cSpellWords.txt ${{ inputs.path }}/.github
+      cd ${{ inputs.path }}
+      set -e
+
+      echo -e "::endgroup::"
+      # Make sure we have all the commands we need.
+      echo -e "${{ env.bashInfo }} fdfind --version $(fdfind --version) ${{ env.bashEnd }}"
+      echo -e "${{ env.bashInfo }} cspell --version $(cspell --version) ${{ env.bashEnd }}"
+
+      # Only reach this line if no errors were hit above
+      echo -e "${{ env.bashPass }} ${{ env.stepName }} ${{ env.bashEnd }}"
+
+  - env:
+      bashPass: \033[32;1mPASSED -
+      bashInfo: \033[33;1mINFO -
+      bashFail: \033[31;1mFAILED -
+      bashEnd:  \033[0m
+      stepName: Spell Checker
+    name: ${{ env.stepName }}
+    id: run-spell-checker
+    working-directory: ${{ inputs.path }}
+    shell: bash
+    run: |
+      # ${{ env.stepName }}
+      echo -e "::group::${{ env.bashInfo }} ${{ env.stepName }} ${{ env.bashEnd }}"
+      export PATH="$GITHUB_ACTION_PATH:$PATH"
+      exitStatus=0
+
+      # Parse the optional inputs
+      args=""
+
+      # fd-find uses -E to exclude a file or directory
+      if [ -n "${{ inputs.exclude-dirs }}" ]; then
+        dirs=" -E "
+        dirs+="${{ inputs.exclude-dirs }}"
+        dirs="${dirs//,/ -E }"
+        args+=" ${dirs}"
+      fi
+
+      # fd-find uses -E to exclude a file or directory
+      if [ -n "${{ inputs.exclude-files }}" ]; then
+        files=" -E "
+        files+="${{ inputs.exclude-files }}"
+        files="${files//,/ -E }"
+        args+=" ${files}"
+      fi
+
+      # fd-find uses -e to exclude a file extension
+      if [ -n "${{ inputs.include-file-types }}" ]; then
+        file_types=" -e "
+        file_types+="${{ inputs.include-file-types }}"
+        file_types="${file_types//,/ -e }"
+        args+=" ${file_types}"
+      fi
+
+      echo -e "${{ env.bashInfo }} Running: fdfind -e c -e h -e md -e txt -e readme ${args} --exec-batch cspell lint --no-progress --language-id C --color --show-context --show-suggestions --no-must-find-files -c cspell.config.yaml ${{ env.bashEnd }}"
+
+      # Wrap in a set +e so Github doesn't stop the spell check from running
+      set +e
+
+      # Find all relevant files, then check them for spelling mistakes
+      fdfind -e c -e h -e md -e txt -e readme ${args} --exec-batch \
+        cspell lint --no-progress --language-id C --color --show-context --show-suggestions --no-must-find-files -c cspell.config.yaml
```

Comment:
Example of how this looks before this change [here](https://github.com/FreeRTOS/FreeRTOS/actions/runs/6477161501/job/17586919705#step:3:1241)
```
 INFO - Running: fdfind -e c -e h -e md -e txt -e readme   -E ethernet -E  drivers --exec cspell lint --language-id C --color --show-context --show-suggestions --no-must-find-files -c cspell.config.yaml 

  1/106 ./ARMv8M/mpu_demo/mpu_demo.c 765.04ms

  2/106 ./ARMv8M/mpu_demo/mpu_demo.h 29.12ms

  3/106 ./ARMv8M/reg_tests/GCC/ARM_CM23/non_secure/reg_test_asm.c 47.69ms

  4/106 ./ARMv8M/reg_tests/GCC/ARM_CM23/non_secure/reg_test_asm.h 17.46ms

  <REST OF OUTPUT>
```

Example of how this looks after the change, when it works, [here](https://github.com/FreeRTOS/CI-CD-Github-Actions/actions/runs/6485828214/job/17612745509#step:10:1251)
```
  INFO - Running: fdfind -e c -e h -e md -e txt -e readme   -E FreeRTOS --exec-batch cspell lint --no-progress --language-id C --color --show-context --show-suggestions --no-must-find-files -c cspell.config.yaml 
  CSpell: Files checked: 600, Issues found: 0 in 0 files
PASSED - Spell Checker 
```